auth_mellon_cache.c

```diff
@@ -465,6 +467,25 @@ void am_cache_update_expires(request_rec *r, am_cache_entry_t *t, apr_time_t exp
     }
 }
 
+/* This function updates the idle timeout timestamp of a session, if the
+ * actual idle timeout is not disabled (-1).
+ *
+ * Parameters:
+ *  request_rec *r           The request we are processing.
+ *  am_cache_entry_t *t      The current session.
+ *  apr_time_t idle_timeout  The new timestamp.
+ *
+ * Returns:
+ *  Nothing.
+ */
+void am_cache_update_idle_timeout(request_rec *r, am_cache_entry_t *t, apr_time_t idle_timeout)
+{
+    /* If idle_timeout timestamp is not -1 (disabled), update the value. */
+    if(t->idle_timeout != -1) {
```

Comment:
Why this check here ?
There are checks about whether the default is -1 in the code that invoke this function.

Maybe either move the logic all in here, or just set the timeout detemined by the caller unconditionally?

It is a bit confusing to have the logic split across.

Reply:
understand

Reply:
done, i have decided to centralize the logic here.